pkg/setting/setting_feature_toggles.go

```diff
@@ -10,65 +13,188 @@ import (
 	"gopkg.in/ini.v1"
 )
 
+type FeatureToggleState string
+
+const (
+	AlphaState   FeatureToggleState = "alpha"   // anything can change without warning
+	BetaState    FeatureToggleState = "beta"    // shape looks right, but still evoloving
+	StableState  FeatureToggleState = "stable"  // stable and may soon become a build-in feature
+	MergedState  FeatureToggleState = "merged"  // feature no longer requires an explicit flag
+	RemovedState FeatureToggleState = "removed" // the feature was not merged and no longer has any effect
+)
+
 var (
 	featureToggleInfo = promauto.NewGaugeVec(prometheus.GaugeOpts{
 		Name:      "feature_toggles_info",
 		Help:      "info metric that exposes what feature toggles are enabled or not",
 		Namespace: "grafana",
 	}, []string{"name"})
+)
+
+type FeatureToggleInfo struct {
+	Id          string             `json:"id"`   // the key listed in grafana.ini
+	Name        string             `json:"name"` // user
+	Description string             `json:"description"`
+	AliasIds    []string           `json:"aliasIds,omitempty"` // ids that will also work (support spelling changes and product renames)
+	State       FeatureToggleState `json:"state,omitempty"`
+	DocsURL     string             `json:"docsURL,omitempty"`
+
+	// Loading state
+	Enabled bool `json:"enabled,omitempty"`
+
+	// Special flags
+	RequiresDevMode    bool `json:"requiresDevMode,omitempty"`    // can not be enabled in production
+	RequiresEnterprise bool `json:"requiresEnterprise,omitempty"` // requires enterprise executable
+	ModifiesDatabase   bool `json:"modifiesDatabase,omitempty"`   // the feature will modify the SQL tables
+	FrontendFlag       bool `json:"frontend,omitempty"`           // the change is only seen in the frontend (can change without restart!)
```

Comment:
Reloading feature without restarting should be supported by both frontend and backend. FGAC is a good example of a feature that could enable at runtime.

Reply:
:+1: